doc/devel/documenting_mpl.rst

```diff
@@ -136,28 +144,54 @@ An example docstring looks like:
         axhline: horizontal line across the axes
         """
 
+See the `~.Axes.hlines` documentation for how this renders.
+
 The Sphinx website also contains plenty of documentation_ concerning ReST
 markup and working with Sphinx in general.
 
 Linking to other code
----------------------
+~~~~~~~~~~~~~~~~~~~~~
 To link to other methods, classes, or modules in Matplotlib you can encase
 the name to refer to in back ticks, for example:
 
 .. code-block:: python
 
   `~matplotlib.collections.LineCollection`
 
-It is also possible to add links to code in Python, Numpy, Scipy, or Pandas.
-Sometimes it is tricky to get external Sphinx linking to work; to check that
-a something exists to link to the following shell command outputs a list of all
-objects that can be referenced (in this case for Numpy)::
+returns a link to the documentation of
+`~matplotlib.collections.LineCollection`.  If we want the full path of the
+class to be shown then we omit the tilde:
+
+.. code-block:: python
+
+  `matplotlib.collections.LineCollection`
+
+to get `matplotlib.collections.LineCollection`.  Its often not
+necessary to fully specify the class hierarchy:
+
+.. code-block:: python
+
+  `~.LineCollection`
+
+links just as well: `~.LineCollection`.
+
+Other packages can also be linked via ``intersphinx``:
+
+.. code-block:: Python
+
+  `numpy.mean`
+
+will return this link: `numpy.mean`.  This works for Python, Numpy, Scipy,
+and Pandas. Sometimes it is tricky to get external Sphinx linking to work; to
+check that a something exists to link to the following shell command outputs
+a list of all objects that can be referenced (in this case for Numpy)::
 
   python -m sphinx.ext.intersphinx 'https://docs.scipy.org/doc/numpy/objects.inv'
 
 
 Function arguments
-------------------
+~~~~~~~~~~~~~~~~~~
+
```

Comment:
may be worth explaining briefly why backquotes doesn't work (arguments don't have an entry)

Reply:
Changed to:

```rst

   Do not describe `argument` like this.  As per the previous section,
   this syntax will attempt to resolve the argument as a link to an object in the
   library.
```

Reply:
will (unsuccessfully) attempt to etc.